src/test/fuzz/script.cpp

```diff
@@ -15,20 +15,36 @@
 #include <script/standard.h>
 #include <streams.h>
 #include <test/fuzz/fuzz.h>
+#include <univalue.h>
 #include <util/memory.h>
 
 void initialize()
 {
     // Fuzzers using pubkey must hold an ECCVerifyHandle.
     static const auto verify_handle = MakeUnique<ECCVerifyHandle>();
+
+    SelectParams(CBaseChainParams::REGTEST);
 }
 
 void test_one_input(const std::vector<uint8_t>& buffer)
 {
     const CScript script(buffer.begin(), buffer.end());
 
     std::vector<unsigned char> compressed;
-    (void)CompressScript(script, compressed);
+    if (CompressScript(script, compressed)) {
+        const unsigned int size = compressed[0];
+        assert(size >= 0 && size <= 5);
+        CScript decompressed_script;
+        const bool ok = DecompressScript(decompressed_script, size, compressed);
+        assert(ok);
```

Comment:
Could also assert the roundtrip `decompressed == script`?

Reply:
I'm afraid not since that assertion is not guaranteed to hold :)

Counter-example:

```
$ xxd -p -r <<< "210220acacacffff6010b472a266d0bd89c13706a4132ccfb16f7c3b9fcbacacacacac" > counter-example
$ src/test/fuzz/script counter-example
…
script: test/fuzz/script.cpp:40: void test_one_input(const std::vector<uint8_t> &): Assertion `script == decompressed_script' failed.
…
```

Reply:
I am pretty sure that if it was not guaranteed to hold, we would be running into consensus failures pretty soon. Please take a look at #18292

Reply:
With #18292 it all makes sense. Thanks!